rfcs/20180827-api-names.md

```diff
@@ -0,0 +1,587 @@
+| tf.saturate_cast | tf.dtypes.saturate_cast |
+| tf.saved_model.builder.SavedModelBuilder | tf.saved_model.SavedModelBuilder |
+| tf.saved_model.constants.ASSETS_DIRECTORY | tf.saved_model.ASSETS_DIRECTORY |
+| tf.saved_model.constants.ASSETS_KEY | tf.saved_model.ASSETS_KEY |
+| tf.saved_model.constants.LEGACY_INIT_OP_KEY | tf.saved_model.LEGACY_INIT_OP_KEY |
+| tf.saved_model.constants.MAIN_OP_KEY | tf.saved_model.MAIN_OP_KEY |
+| tf.saved_model.constants.SAVED_MODEL_FILENAME_PB | tf.saved_model.SAVED_MODEL_FILENAME_PB |
+| tf.saved_model.constants.SAVED_MODEL_FILENAME_PBTXT | tf.saved_model.SAVED_MODEL_FILENAME_PBTXT |
+| tf.saved_model.constants.SAVED_MODEL_SCHEMA_VERSION | tf.saved_model.SAVED_MODEL_SCHEMA_VERSION |
+| tf.saved_model.constants.VARIABLES_DIRECTORY | tf.saved_model.VARIABLES_DIRECTORY |
+| tf.saved_model.constants.VARIABLES_FILENAME | tf.saved_model.VARIABLES_FILENAME |
+| tf.saved_model.loader.load | tf.saved_model.load |
+| tf.saved_model.loader.maybe_saved_model_directory | tf.saved_model.maybe_saved_model_directory |
+| tf.saved_model.main_op.main_op_with_restore | tf.saved_model.main_op_with_restore |
+| tf.saved_model.signature_constants.CLASSIFY_INPUTS | tf.saved_model.CLASSIFY_INPUTS |
+| tf.saved_model.signature_constants.CLASSIFY_METHOD_NAME | tf.saved_model.CLASSIFY_METHOD_NAME |
+| tf.saved_model.signature_constants.CLASSIFY_OUTPUT_CLASSES | tf.saved_model.CLASSIFY_OUTPUT_CLASSES |
+| tf.saved_model.signature_constants.CLASSIFY_OUTPUT_SCORES | tf.saved_model.CLASSIFY_OUTPUT_SCORES |
+| tf.saved_model.signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY | tf.saved_model.DEFAULT_SERVING_SIGNATURE_DEF_KEY |
+| tf.saved_model.signature_constants.PREDICT_INPUTS | tf.saved_model.PREDICT_INPUTS |
+| tf.saved_model.signature_constants.PREDICT_METHOD_NAME | tf.saved_model.PREDICT_METHOD_NAME |
+| tf.saved_model.signature_constants.PREDICT_OUTPUTS | tf.saved_model.PREDICT_OUTPUTS |
+| tf.saved_model.signature_constants.REGRESS_INPUTS | tf.saved_model.REGRESS_INPUTS |
+| tf.saved_model.signature_constants.REGRESS_METHOD_NAME | tf.saved_model.REGRESS_METHOD_NAME |
+| tf.saved_model.signature_constants.REGRESS_OUTPUTS | tf.saved_model.REGRESS_OUTPUTS |
+| tf.saved_model.signature_def_utils.build_signature_def | tf.saved_model.build_signature_def |
+| tf.saved_model.signature_def_utils.classification_signature_def | tf.saved_model.classification_signature_def |
+| tf.saved_model.signature_def_utils.is_valid_signature | tf.saved_model.is_valid_signature |
+| tf.saved_model.signature_def_utils.predict_signature_def | tf.saved_model.predict_signature_def |
+| tf.saved_model.signature_def_utils.regression_signature_def | tf.saved_model.regression_signature_def |
+| tf.saved_model.tag_constants.GPU | tf.saved_model.GPU |
+| tf.saved_model.tag_constants.SERVING | tf.saved_model.SERVING |
+| tf.saved_model.tag_constants.TPU | tf.saved_model.TPU |
+| tf.saved_model.tag_constants.TRAINING | tf.saved_model.TRAINING |
```

Comment:
The lack of tag_constants here makes the meaning of these less clear. Is the primary goal reducing verbosity? Maybe .tags., or GPU_TAG even? Not sure.

Reply:
Pasting @martinwicke 's response here to preserve the coherence of the thread: The primary reason is to reduce incidental complexity. This is particular
important for `signature_def_utils` and `signature_constants`, which are
very redundant. For tag_constants, how about making it a class, not a
module? tf.saved_model.Tags.GPU (and yeah, shorter, as you suggest as well)?

Reply:
To be perfectly honest, I am one of the world-experts in these sets of constants at this point, and I still forget what the difference between tag_constants and signature_constants is, and which is used where. Let me think about these a little, and see if I have a better suggestion.